### ISSUE TITLE
Installing the library

### ISSUE THREAD

**Issue by ghost** [JeffHoogland/python-elm-extensions#7]
Hey, I was wondering how I should install the library. I see no setup.py, so should I assume that I'd have to put the .py files in `/usr/lib64/python3.3`?

**Comment by JeffHoogland** [comment]
You will want to put it in any path your python can import from. Fair warning I haven't tested my last few commits against python 3, so please report any issues you might find if you use Git instead of a release.

**Comment by Annwenn** [comment]
so, I assume all of the .py files (excluding test_* and __init__.py)?

**Comment by JeffHoogland** [comment]
Basically just copy the elmextensions directory and leave everything else
out.

**Comment by Annwenn** [comment]
Thanks :)

**Comment by wltjr** [comment]
I create one on the fly in my gentoo ebuild
```
echo "#!/usr/bin/env python

from distutils.core import setup

setup(name='${PN}',
        version='${PV}',
        description='${DESCRIPTION}',
        url='${HOMEPAGE}',
        packages=['elmextensions'],
        py_modules=['sortedlistother.sortedlist'],
)" > setup.py
```

[Link to ebuild](https://github.com/Obsidian-StudiosInc/os-xtoo/blob/master/dev-python/python-elm-extensions/python-elm-extensions-9999.ebuild)

**Comment by rbtylee** [comment]
I am closing this as this is a non-issue. It would be nice to add a 'proper' setup.py file but for now it is hardly a priority. Any python-developer would know how to use this code, see Jeff's comments above. And all users of Bodhi have it preinstalled, the python2 versions and I am adding to Bodhi and planning on maintaining the python3 version.

**Comment by wltjr** [comment]
Not everyone is a python developer. It is just something to make the package friendly for end users and packagers. This is a dependency of some other Python EFL applications. Someone may not be installing this to do python development. I consider it good house keeping and being a good upstream. Any good project ships with means to build it, some build system and related files. Either way moot to me, but was something I ran into and I had to spend a bit of time learning about setup.py. There maybe a Gentoo eclass to provide such not sure. I do not package many python apps, but from the ones I have. This was the only to lack a setup.py, to my knowledge. That it is used by other projects, it likely should have one. If setup with a CI could generate on the fly when tagged given that it seems to want versions and stuff. Again not sure I am not a python developer. First time I had to make a setup.py.

**Comment by JeffHoogland** [comment]
This isn't an end user application. It is a library for writing other pieces of software.

**Comment by wltjr** [comment]
I know, it is also a dependency of some other Python EFL applications... I have no interest in this. I only had to touch it for other stuff, presently just epack, but also epad and some others. Anyone who wants to use those end user applications needs this library....

I did not open the bug. I simply provided a sample setup.py from the ebuild I made to generate one on the fly. I already addressed this issue. I come across unfriendly upstreams all the time. But you know it is easier to make a setup.py, than comment on this issue :)
